File: python/smoke.py
"""Quick client smoke — polite delay on."""

from __future__ import annotations

import json

from pabc_card_offers_docs import PabcCardOffersDocsClient


def json_preview(data: object, limit: int = 200) -> str:
    try:
        return json.dumps(data, ensure_ascii=False)[:limit]
    except TypeError:
        return str(data)[:limit]


with PabcCardOffersDocsClient(default_delay_seconds=1.0) as client:
    print("smoke", client.slug, "->", "card_offers_js")
    data = client.card_offers_js()
    preview = data[:200] if isinstance(data, str) else json_preview(data)
    print("ok", preview)

File: typescript/examples/smoke.ts
import { PabcCardOffersDocsClient } from "../src/index.js";

/**
 * Live smoke — polite delay on. Run: npm run smoke
 * May fail if upstream is down / blocks this environment.
 */
async function main() {
  const client = new PabcCardOffersDocsClient({ defaultDelayMs: 1000 });
  console.log("smoke", PabcCardOffersDocsClient.slug, "->", "cardOffersJs");
  const data = await client.cardOffersJs();
  const preview = typeof data === "string" ? data.slice(0, 200) : JSON.stringify(data)?.slice(0, 200);
  console.log("ok", preview);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
